ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/FastSyncActions.java

```diff
@@ -177,13 +184,55 @@ public ChainDownloader createChainDownloader(
 
   private CompletableFuture<FastSyncState> downloadPivotBlockHeader(final Hash hash) {
     LOG.debug("Downloading pivot block header by hash {}", hash);
-    return RetryingGetHeaderFromPeerByHashTask.byHash(
-            protocolSchedule,
-            ethContext,
-            hash,
-            pivotBlockSelector.getMinRequiredBlockNumber(),
-            metricsSystem)
-        .getHeader()
+    CompletableFuture<BlockHeader> blockHeaderFuture;
+    if (syncConfig.isPeerTaskSystemEnabled()) {
+      blockHeaderFuture =
+          ethContext
+              .getScheduler()
+              .scheduleServiceTask(
+                  () -> {
+                    GetHeadersFromPeerTask task =
+                        new GetHeadersFromPeerTask(
+                            hash,
+                            pivotBlockSelector.getMinRequiredBlockNumber(),
+                            1,
+                            0,
+                            GetHeadersFromPeerTask.Direction.FORWARD,
+                            protocolSchedule);
+                    PeerTaskExecutorResult<List<BlockHeader>> taskResult =
+                        peerTaskExecutor.execute(task);
+                    if (taskResult.responseCode() == PeerTaskExecutorResponseCode.NO_PEER_AVAILABLE
+                        || taskResult.responseCode()
+                            == PeerTaskExecutorResponseCode.PEER_DISCONNECTED) {
```

Comment:
We should try another peer if the one we tried got disconnected ...

Reply:
The PeerTaskExecutor would have already tried all available peers, since we set retries to `Integer.MAX_VALUE`